docs/assignment2.md

```diff
@@ -1 +1,233 @@
+### *Models* package  
+  
+This package implements 5 distinct classes: **Game**, **League**, **Team**, **Player** and **Match**, described below.  
+  
+#### **Game**  
+  
+This class is responsible with managing the data of the game and coordinating actions across the system.
+  
+Attributes 
+  
+- *allPlayers* (type: ArrayList<Player&gt;): list of all user-added players  
+- *allTeams* (type: ArrayList<Team&gt;): list of all user-created teams  
+- *league* (type: League): user-created league  
+  
+Operations
+  
+- *void addPlayers(String pathToFile)* : method for adding new players to the *allPlayers* arraylist; parameter specifies the path to the CSV file, as per **FR1**  
+  
+- *void addTeam(Player[] players, String teamName)*: method for adding a team to the *allTeams* arraylist; parameters are the same as the ones required by the **Team** constructor.  
+  
+- *void createLeague(Queue<Team&gt; schedule)*: method for defining the league to be played; parameter is a list of teams ordered by their playing order, e.g.  a round with matches (Team1 vs. Team2) and (Team3 vs. Team4) would be represented as {Team1, Team2, Team3, Team4}.
+- *HashMap<Integer, ArrayList<Team&gt;> getLeagueResults()*: returns the winners of *league* in the original format of the **League** class (described below)
+- *ArrayList<Team&gt; generateLeaderboard()*: returns a sublist of first 10 teams in *allTeams*  sorted by their number of wins (**Team**.wins)
+- *ArrayList<String&gt; getAllTeamNames()*: returns the team names of teams in *allTeams*
+- *Team getTeamByName(String name)*: returns the team from *allTeams* with the given *name*
+- *void removeTeam(String teamName)*: removes the team with the given *teamName* from *allTeams*
+- *void removePlayer(String playerName)*: removes the player with the given *playerName* from *allPlayers*
+
+Associations
+
+As can be observered from the class diagram, the **Game** class has composition relations with **Team**, **Player** and **League**, as it is resposible with creating and storing these other objects. While it can store an indefinite number of teams and players, a game can only have one league, as a user can not play multiple leagues at one time.  It also has associations with the GUI classes, which intercat with **Game** to either follow up on user actions, e.g. forming a new team, or get the data to be displayed to the user, e.g. the leaderboard.
+
+#### **Player**
+
+Class representing a single player. Only exists within a **Game** and is created from user input.
+
+Attributes 
+
+- *name* (type: String): player's name
+- *score* (type: int): player's score
+
+Operations
+
+- *String getName()*: returns *name*
+- *int getScore()* : returns *score*
+
+Associations
+
+All **Player** objects are contained within **Game**. A team is an aggregation of exactly 8 **Player** objects. 
```

Comment:
**[INCONSISTENCY}** In Assignment 1, F3 Team formation states that teams are formed of 11 players, I don't see the incentive behind this change ?